src/commonMethods/commonFunctions.js

```diff
@@ -0,0 +1,353 @@
+
+  // wait for the promise to resolve
+  let result = await bufferData
+  this.$data.write('screenshotResult', result)
+}
+
+/**
+ * This function is used to kill the process
+ * @param process
+ */
+export const killProcess = function(process) {
+  exec({ cmd: `killall -9 ${process}` }, err => {
+    return true
+  })
+}
+
+/**
+ * This function is used to start the framework
+ */
+export const startFramework = function() {
+  exec({ cmd: 'nohup WPEFramework WPEProcess -b &', cbWhenStarted: true }, err => {
+    return true
+  })
+}
+
+/**
+ * This function is used to create the server
+ * @param requestFunction
+ */
+export const startHttpServer = function(requestFunction) {
+  let data = this.$data.read('app')
+  let server = http.createServer(function(req, res) {
+    res.writeHead(200, { 'Content-Type': 'text/html' })
+    res.write(data)
+    res.end()
+  })
+  server.listen(() => {
+    this.$data.write('port', server.address().port)
+  })
+}
+
+/**
+ * This function is used to match the IP range of local server and the RbPi
+ * @returns {*}
+ */
+export const matchIpRange = function() {
+  let ip = constants.host
+  let localInterfaces = os.networkInterfaces()
+  let parsedIp = ip.split('.')
+  let interfaceList = Object.keys(localInterfaces)
+  for (var i = 0; i < interfaceList.length; i++) {
+    var interface1 = localInterfaces[interfaceList[i]]
+
+    for (var k = 0; k < interface1.length; k++) {
+      if (interface1[k].family === 'IPv6') continue //not supporting IPv6, this shouldn't be called in IPv6 mode as there is no need for private/lan ranges anymore
+      if (interface1[k].address !== undefined) {
+        var parsedLocalIp = interface1[k].address.split('.')
+
+        if (
+          parsedIp[0] === parsedLocalIp[0] &&
+          parsedIp[1] === parsedLocalIp[1] &&
+          parsedIp[2] === parsedLocalIp[2]
+        ) {
+          return interface1[k].address
+        }
+      }
+    }
+    if (i === interfaceList.length - 1)
+      throw new Error('Could not determine IP network of the server that belongs to: ' + ip)
+  }
+}
+
+/**
+ * This function is used to restart the framework
+ */
+export const restartFramework = function() {
+  return this.$sequence([
+    () => killProcess('WPEFramework'),
+    () => killProcess('WPEProcess'),
+    () => startFramework(),
+  ])
+}
+
+/**
+ * This function performs below operations on WebKitBrowser Plugin
+ *  - Deactivate
+ *  - Activate
+ *  - Resume
+ */
+export const webKitBrowserOps = function() {
+  return this.$sequence([
+    () => pluginDeactivate.call(this, constants.webKitBrowserPlugin),
+    () => pluginActivate.call(this, constants.webKitBrowserPlugin),
+    () => webKitBrowserActions.call(this, constants.resume),
+  ])
+}
+
+/**
+ * This function is used to get Plugin Info
+ * @param plugin
+ * @returns {Promise<AxiosResponse<any>>}
+ */
+export const getPluginInfo = function(plugin) {
+  return this.$http
+    .get(`http://${constants.host}:80/Service/${plugin}`)
```

Comment:
This is a deprecated REST API, should not be used anymore and should have a JSONRPC equivalent.